### ISSUE TITLE
OfflineCompression error if one uses hook_output

### ISSUE THREAD

**Issue by shredding** [torchbox/wagtail#2054]
I'm not sure if it's a bug or something that requires advanced config on my side, but since it took me a few hours to track this down I want to bring it up to discussion:

https://github.com/torchbox/wagtail/blob/master/wagtail/wagtailadmin/templates/wagtailadmin/pages/_editor_js.html#L28

We have a simple hook that adds some custom staff to the editor.

```python
@hooks.register('insert_editor_js')
def editor_js():
    # SOME STUFF HERE THAT DOES NICE THINGS ...
    <script type="text/javascript">
        window.PageEditorEnhancements = {{
            translations: {{
                confirmUnsavedChanges: "{confirm_unsaved_changes}"
            }}
        }};
    </script>
    """.format(
        confirm_unsaved_changes=_(
            'It looks like you have been editing something. If you leave before saving, your changes will be lost.'
        )
    )
```

Note that we have this tiny translatable "confirm_unsaved_changes" that one of our product managers decided to translate today into german. BOOM, desaster strikes because the django-compressor manifest.json key created during backend is different then the one created for german users. This ended up in the admin being unusable and I was very close to giving up on that one :)

However - one could argue that this is something that should go within `COMPRESS_OFFLINE_CONTEXT` but it's cumbersome, because one does not expect the hook to compress stuff (or one does not understand django-compressor on an expert level and feels that django-compressor is notouriously hard to debug).

Long story short: I'd propose to move the given line *below* the `<compress>`-Block.

**Comment by gasman** [comment]
Hi @shredding,

It's probably not feasible to move this now, since there's a lot of existing code - both within Wagtail and outside - where applying compression *is* the right thing to do. In particular, anything that's using django-compressor's preprocessor support (to compile from Coffeescript to JS, say) will break.

I would suggest adding a new hook (possibly named `insert_editor_request_html`, since there's no longer anything JS-specific about it) which exists outside of the `{% compress %}` block, and receives a `request` parameter. The idea would be that any HTML declarations that could vary on a per-request basis would be inserted through this hook, rather than through `insert_editor_js`.

**Comment by shredding** [comment]
Uh, you are right. Adding that additional block would be awesome, as one can just put all the logic in the compress block. Maybe we could add a note to the docs (but to be honest I did not look into the docs either).

**Comment by gasman** [comment]
If I'm not mistaken, this is no longer necessary now that we've dropped django-compressor.
